### PR TITLE
DYN-5007 Group Collapsed Hiding Wires

### PR DESCRIPTION
**Pull request by filipeotero**
### Purpose

It's included in this PR the fix for the bug that the output wires are disappearing when the group is collapsed.
The conditions to get the ports and wires to be shown after the group collapsed were adapted for the situation when there are two connections going out from the same outport. One wire is connected inside the group and the other outside.

showhire

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang @zeusongit @reddyashish

### CODE REVIEW

**Comment by zeusongit** [comment]
As per the gif attached the wires are updated inside the group when the group is moved/dragged?

**Comment by reddyashish** [comment]
Yes, it looks like the connectors are not in place when the group is expanded. You might want to call UpdateProxyPortsPosition() from AnnotationViewModel to recalibrate the port locations, if the connectors are set to correct port owners.

**Comment by QilongTang** [comment]
@reddyashish @zeusongit How is the updated PR?

**Comment by reddyashish** [comment]
Changes look fine to me. @filipeotero Can you attach a latest gif with the new changes?

**Comment by filipeotero** [comment]
> Changes look fine to me. @filipeotero Can you attach a latest gif with the new changes?

Hi, it looks like the UpdateProxyPortsPosition didn't have an effect because the wires were still in the wrong position. I created a new function that passes through the connectors and redraws the wires. It's in the next commit:

updateWires

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 1]
```diff
@@ -889,6 +889,24 @@ private void CollapseConnectors()
             }
         }
 
+        private void RedrawConnectors()
```
[the remark this answers is not in the document]
Comment (reddyashish) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:892]: Can we redraw the connectors for the group that is collapsed/expanded only?

[the remark this answers is not in the document]
Comment (filipeotero) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:892]: Hi @reddyashish, in this method we just get the nodes inside the group. From the workspace, we retrieve the ViewModel by searching for its GUID.

**Comment by filipeotero** [comment]
> According to the gif, when the group is collapsed, it looks the CBN connectors are connected to a point between the 2 CBN ports. Can you look into that.

Thanks for noticing that! I've updated the code and now the point's position is also being refreshed. 
collapseGroupWires

**Comment by QilongTang** [comment]
@reddyashish How do you feel about the updated PR?

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (filipeotero) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:892]: By debugging I confirmed that is getting the nodes of the group that was collapsed:
examplegroups

**Review by reddyashish — APPROVED** [review]

**Comment by reddyashish** [comment]
Merging this Aaron.

**Pull request merged**